cloud/utils/retry.go

```diff
@@ -0,0 +1,63 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package utils
+
+import (
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+// RetryOnFailure allows to retry a lambda up to a failure condition depending on the implementation
+type RetryOnFailure interface {
+	Run() error
+}
+
+// exponentialBackOffRetrier implements RetryOnFailure with an exponential back off on retrying funcToRetry
+type exponentialBackOffRetrier struct {
+	funcToRetry func() error
+	backoff     backoff.BackOff
+}
+
+// NewRetrier returns a retrier using an exponential backoff to retry funcToRetry.
+func NewRetrier(funcToRetry func() error) RetryOnFailure {
+	expBackoff := backoff.NewExponentialBackOff()
+	expBackoff.InitialInterval = 500 * time.Millisecond
```

Comment:
Open to discussion on these parameters, and we can refine them later on if we need to